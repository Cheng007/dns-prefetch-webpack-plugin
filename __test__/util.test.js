const fs = require("fs");
const path = require("path");

const { expect, describe, it, beforeEach } = require("@jest/globals");
const {
  getAllDomain,
  getPrefetchHtmlString,
  getAllScript,
  getAllLink,
} = require("../lib/util");

describe("util", () => {
  let html;
  beforeEach(async () => {
    html =
      html ||
      (await fs.promises.readFile(
        path.resolve(__dirname, "./template/base.html"),
        "utf8",
      ));
  });

  it("get all script", () => {
    const scripts = getAllScript(html);
    const expeted = [
      '<script defer src="main.bundle.js"></script>',
      '<script src="https://zhongan.com/path/test.js"></script>',
    ];
    expect(scripts).toEqual(expeted);
  });

  it("get all link", () => {
    const scripts = getAllLink(html);
    const expeted = ['<link href="http://example.com/path.css" />'];
    expect(scripts).toEqual(expeted);
  });

  it("get all domain", () => {
    const domains = getAllDomain(html);
    expect(domains).toHaveLength(2);
    expect(domains).toEqual(
      expect.arrayContaining(["example.com", "zhongan.com"]),
    );
  });

  describe("dns-prefetch", () => {
    it("with custom domains", async () => {
      const customDomains = ["a.com", "b.com"];
      const dnsPrefetchHtml = getPrefetchHtmlString(html, customDomains);
      expect(dnsPrefetchHtml).toMatchSnapshot();
    });

    it("without custom domain", () => {
      const dnsPrefetchHtml = getPrefetchHtmlString(html);
      expect(dnsPrefetchHtml).toMatchSnapshot();
    });
  });
});
