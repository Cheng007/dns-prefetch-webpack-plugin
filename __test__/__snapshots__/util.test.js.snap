// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util dns-prefetch exist dns prefetch link 1`] = `
"<!doctype html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>Webpack App</title>
    <link href="http://example.com/path.css" />
    <link rel="dns-prefetch" href="https://example1.com" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <script defer src="main.bundle.js"></script>
    <script src="https://zhongan.com/path/test.js"></script>
    <script src="https://example1.com/path/test.js"></script>
  
<link rel="dns-prefetch" href="//zhongan.com" />
<link rel="dns-prefetch" href="//example.com" />

</head>
  <body>
    hell world
  </body>
</html>
"
`;

exports[`util dns-prefetch with custom domains 1`] = `
"<!doctype html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>Webpack App</title>
    <link href="http://example.com/path.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <script defer src="main.bundle.js"></script>
    <script src="https://zhongan.com/path/test.js"></script>
  
<link rel="dns-prefetch" href="//a.com" />
<link rel="dns-prefetch" href="//b.com" />

</head>
  <body>
    hell world
  </body>
</html>
"
`;

exports[`util dns-prefetch without custom domain 1`] = `
"<!doctype html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>Webpack App</title>
    <link href="http://example.com/path.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <script defer src="main.bundle.js"></script>
    <script src="https://zhongan.com/path/test.js"></script>
  
<link rel="dns-prefetch" href="//zhongan.com" />
<link rel="dns-prefetch" href="//example.com" />

</head>
  <body>
    hell world
  </body>
</html>
"
`;
